azure-iot-device/azure/iot/device/iothub/abstract_clients.py

```diff
@@ -11,9 +11,15 @@
 import logging
 import os
 import io
-from . import auth
 from . import pipeline
 
+# TODO: different package?
+from azure.iot.device.common.auth import connection_string as cs
```

Comment:
>cs [](start = 62, length = 2)

may be import as connection and token...just to make more meaningful in code.

Reply:
I have mixed feelings about renaming to cs and st.  Yes, it's shorter, but this is the only place we do this and I don't want it to get out of control.

---
In reply to: [426962229](https://github.com/Azure/azure-iot-sdk-python/pull/555#discussion_r426962229) [](ancestors = 426962229)

Reply:
So for context, the reason that we do this is that `connection_string` and `sastoken` are kind of abnormal modules (for us, not in general) in that there are associated values in the modules, such as the constants. Thus we need to import the whole module, not just the objects in it.

The problem is that `connection_string` easily gets overloaded, and I didn't want to have to worry about accidentally doing that, so I renamed the package to be something that is out of the way and not likely to be used in a variable name by accident.

---
In reply to: [427502200](https://github.com/Azure/azure-iot-sdk-python/pull/555#discussion_r427502200) [](ancestors = 427502200,426962229)

Reply:
I'm OK with that,  I've seen that problem before and never came up with a better answer.  I wouldn't be offended if you left this as-is

---
In reply to: [427504005](https://github.com/Azure/azure-iot-sdk-python/pull/555#discussion_r427504005) [](ancestors = 427504005,427502200,426962229)